test/NuGet.Core.FuncTests/NuGet.Packaging.FuncTest/SigningTests/SignatureTrustAndValidityVerificationProviderTests.cs

```diff
@@ -147,12 +149,108 @@ public SignatureTrustAndValidityVerificationProviderTests(SigningTestFixture fix
             }
         }
 
+        private static FileInfo CreatePackageFile(TestDirectory testDirectory)
+        {
+            var packageContext = new SimpleTestPackageContext();
+            var packageFile = new FileInfo(Path.Combine(testDirectory, "Package.nupkg"));
+
+            using (var readStream = packageContext.CreateAsStream())
+            using (var writeStream = packageFile.OpenWrite())
+            {
+                readStream.CopyTo(writeStream);
+            }
+
+            return packageFile;
+        }
+
+        private static SignatureContent CreateSignatureContent(FileInfo packageFile)
+        {
+            using (var stream = packageFile.OpenRead())
+            using (var hashAlgorithm = HashAlgorithmName.SHA256.GetHashProvider())
+            {
+                var hash = hashAlgorithm.ComputeHash(stream, leaveStreamOpen: false);
+
+                return new SignatureContent(SigningSpecifications.V1, HashAlgorithmName.SHA256, Convert.ToBase64String(hash));
+            }
+        }
+
+        private static SignedCms CreateSignature(SignatureContent signatureContent, X509Certificate2 certificate)
+        {
+            var cmsSigner = new CmsSigner(certificate);
+
+            cmsSigner.DigestAlgorithm = HashAlgorithmName.SHA256.ConvertToOid();
+            cmsSigner.IncludeOption = X509IncludeOption.WholeChain;
+
+            var contentInfo = new ContentInfo(signatureContent.GetBytes());
+            var signedCms = new SignedCms(contentInfo);
+
+            signedCms.ComputeSignature(cmsSigner);
+
+            return signedCms;
+        }
+
+        private static async Task<FileInfo> SignPackageFileAsync(TestDirectory directory, FileInfo packageFile, X509Certificate2 certificate)
```

Comment:
Again, you can use `SignedArchiveTestUtility.CreateSignedPackageAsync` or add appropriate method there to allow reuse..